sqlglot/dialects/postgres.py

```diff
@@ -621,3 +627,12 @@ def datatype_sql(self, expression: exp.DataType) -> str:
                     return f"{self.expressions(expression, flat=True)}[{values}]"
                 return "ARRAY"
             return super().datatype_sql(expression)
+
+        def cast_sql(self, expression: exp.Cast, safe_prefix: t.Optional[str] = None) -> str:
+            this = expression.this
+
+            # Postgres casts DIV() to decimal for transpilation but when roundtripping it's superfluous
+            if isinstance(this, exp.IntDiv) and expression.to.is_type(exp.DataType.Type.DECIMAL):
+                return self.sql(this)
```

Comment:
I think we need to slightly change this condition so that it's `expression.to == exp.DataType.build("decimal")`, otherwise you'll get rid of a cast to `DECIMAL` that includes a scale / precision.

Reply:
You mean removing a user-inputted cast such as `CAST(DIV(...) AS DECIMAL(x, y))`? 

Can't we assume that the user cast will also wrap our generated one, resulting in `CAST(CAST(DIV(...) AS DECIMAL) AS DECIMAL(X, Y))`? So the condition will hold only for our generated cast afaict

Reply:
Actually this could be a problem coming from _other_ dialects I think

Reply:
This is valid DuckDB:

```
SELECT CAST(5 // 3 AS DECIMAL(5, 3))
```

But without building the decimal type, we get rid of the cast. Will switch to building

Reply:
Right